src/dataflow/src/logging/timely.rs

```diff
@@ -387,3 +410,26 @@ pub fn construct<A: Allocate>(
 
     traces
 }
+
+/// Discard all of the records in `c` that `logic` doesn't care about (return).
+fn thin_collection<G, D, R, F>(
+    c: Collection<G, D, R>,
+    delay: std::time::Duration,
+    mut logic: F,
```

Comment:
the compiler forced me to declare this as mutable but I don't really understand why

Reply:
To invoke a `FnMut` you need the function to be mutable. Think of it as a struct with some fields, and `FnMut` means "call with `&mut self`").

Reply:
I was confused because the timely sink implementation takes a FnMut but that wasn't declared as mutable